### PR TITLE
Update `pgbouncer` docs

### PR DESCRIPTION
**Pull request by biw**
Due to https://github.com/prisma/prisma/issues/21531, the currect docs are incorrect regarding setting up `pgbouncer` with `prisma`. This updates the docs to reflect the correct info.

### CODE REVIEW

**Comment by JelteF** [comment]
LGTM

**Comment by gniting** [comment]
@jharrell is there an update on this in terms of getting it merged?

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx:33]: I suggest we keep this for users using previous versions of PgBouncer.

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx:23]: I suggest we keep this for users using previous versions of pgbouncer.

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx]: I suggest keep the discarded changes in the PR and add this in as a note:

```suggestion
:::note

You no longer need to set `pgbouncer=true` if you're using [PgBouncer `1.21.0`](https://github.com/prisma/prisma/issues/21531#issuecomment-1919059472) and above.

:::
```

**Comment by ankur-arch** [comment]
Hey @biw 👋 ,

Thanks for raising the PR! I left some suggestions and we can merge this in after those are resolved.

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (biw) [reply content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx]: Thanks for the review, @ankur-arch! I don't think this is the right edit since "You no longer need to set" gives the impression that it doesn't hurt anything to have `pgbouncer=true` set on the connection string. However, from the linked issue ([#21531](https://github.com/prisma/prisma/issues/21531)):

>> So to recap:
>> 
>> PgBouncer <1.21.0 => Needs pgbouncer=true to work correctly
>> PgBouncer 1.21.0, max_prepared_statements = 0 => Needs pgbouncer=true to work correctly (same behavior as <1.21.0)
>> PgBouncer 1.21.0, max_prepared_statements > 0 => Will fail with > pgbouncer=true (because of https://github.com/prisma/prisma/issues/21635)
>> PgBouncer 1.21.0, max_prepared_statements > 0 => Should work fine without pgbouncer=true (still confirming)
>> PgBouncer >1.21.0 (future release) => Should work fine with and without pgbouncer=true
>> Does that match what you are seeing @​MorenoMdz?
>
> Yes, with a couple things to add.
>
> Our cloud provider has max_prepared_statements set to 100 vs the default of 0.
Having clients with different configurations, one with pgbouncer=true and others without it, will cause pgbouncer to error out on some of the prepared statements requests.

\- From https://github.com/prisma/prisma/issues/21531#issuecomment-1794661977


@JelteF also mentions (https://github.com/prisma/prisma/issues/21531#issuecomment-1919059472):

> I released PgBouncer 1.22.0 just now, which includes the support for DEALLOCATE ALL. So with that release using pgbouncer=true won't cause breakage anymore. But `pgbouncer=true` is not needed anymore and since 1.21.0 and it is slower, so it's recommended to remove it from your connection strings.

I personally tested (albeit N of 1) `pgbouncer=true` with v1.23.1 and IIRC saw a double-digit percentage drop in query performance with a system doing 1000s of prisma transactions per second.


I can edit the PR to include the above matrix of info, but then we would need to add more of a description to the section "Set PgBouncer `max_prepared_statements` > 0" to address various different PgBouncer versions. I'm advocating for only documenting version 1.22.0  (released 2024-01-31) and newer. I imagine having docs for multiple versions/configurations of PgBouncer would add confusion for the vast majority of developers, but can edit it if you disagree.

A bigger question for you and the dev team at prisma is: 

"Should `?pgbouncer=true` be deprecated given that @JelteF, one of the maintainers of PgBouncer recommends not using it?"

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (biw) [reply content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx:23]: I'm assuming https://github.com/prisma/docs/pull/6346/files#r1866103335 supersede this comment 😄

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [reply content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx]:
```suggestion
:::warning
We recommend **not** setting `pgbouncer=true` in the database connection string if you're using [PgBouncer `1.21.0`](https://github.com/prisma/prisma/issues/21531#issuecomment-1919059472) or later.
:::
```

**Comment by ankur-arch** [comment]
>I imagine having docs for multiple versions/configurations of PgBouncer would add confusion for the vast majority of developers, but can edit it if you disagree.

That's a fair point. However, users using older versions of PgBouncer will run into a different confusion 😄 . To simplify, I think we can also add a note asking users to use the latest version of PgBouncer.


> A bigger question for you and the dev team at prisma is:
> "Should ?pgbouncer=true be deprecated given that @JelteF, one of the maintainers of PgBouncer recommends not using it?"

That's a very good question @biw and has been brought to the attention of our team. I'll follow-up on our discussions here 👍 . 

Sidenote: this PR will close https://github.com/prisma/docs/issues/5708.

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx:22]:
```suggestion
### Add `pgbouncer=true` to the connection URL
```

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx]:
```suggestion
### Configure `max_prepared_statements` in PgBouncer to be greater than zero
```

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx]:
```suggestion
Prisma uses prepared statements, and setting [`max_prepared_statements`](https://www.pgbouncer.org/config.html) to a value greater than `0` enables PgBouncer to use those prepared statements.
```

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx:22]:
```suggestion
### Add `pgbouncer=true` to the connection URL

```

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx:28]:
```suggestion

To use Prisma Client with PgBouncer, add the `?pgbouncer=true` flag to the PostgreSQL connection URL:

```shell
postgresql://USER:PASSWORD@HOST:PORT/DATABASE?pgbouncer=true
``` ```

:::info
`PORT` specified for PgBouncer pooling is sometimes different from the default `5432` port. Check your database provider docs for the correct port number.
::: 

```

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx]:
```suggestion
### Add `pgbouncer=true` for PgBouncer versions below `1.21.0`
```

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx:32]:
```suggestion
```
```

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx:21]:
```suggestion
Transaction mode offers a connection for every transaction – a requirement for the Prisma Client to work with PgBouncer.

```

File: content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx
Comment (ankur-arch) [inline content/200-orm/200-prisma-client/000-setup-and-configuration/050-databases-connections/200-pgbouncer.mdx]:
```suggestion
:::info
`PORT` specified for PgBouncer pooling is sometimes different from the default `5432` port. Check your database provider docs for the correct port number.
:::
```

**Review by ankur-arch — APPROVED** [review]

**Comment by jharrell** [comment]
Thanks for your patience @biw and @JelteF ! Excited to get this merged 😊

**Pull request merged**
